src/controllers/scripting/legacy/controllerscriptenginelegacy.h

```diff
@@ -42,23 +50,68 @@ class ControllerScriptEngineLegacy : public ControllerScriptEngineBase {
     void setSettings(
             const QList<std::shared_ptr<AbstractLegacyControllerSetting>>& settings);
 
+#ifdef MIXXX_USE_QML
+    void setLibraryDirectories(const QList<LegacyControllerMapping::QMLModuleInfo>& scripts);
+    void setInfoScrens(const QList<LegacyControllerMapping::ScreenInfo>& scripts);
+    void setResourcePath(const QString& resourcePath) {
+        m_resourcePath = resourcePath;
+    }
+
+  private slots:
+    void handleScreenFrame(
+            const LegacyControllerMapping::ScreenInfo& screeninfo,
+            const QImage& frame,
+            const QDateTime& timestamp);
+
+  signals:
+    /// Emitted when a screen has been rendered
+    // TODO (XXX) Move this signal in ControllerScriptEngineBase when ScreenInfo
+    // isn't tight to LegacyControllerMapping anymore
```

Comment:
if you really plan to do that, can you file an issue and link it here so we remember to follow up?

Reply:
I've created [an issue](https://github.com/mixxxdj/mixxx/issues/13203), although it isn't trivial as `ControllerScriptModuleEngine` still would need quite some work.